view/handlers/post.go

```diff
@@ -0,0 +1,235 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/serhiihuberniuk/blog-api/models"
+	viewmodels "github.com/serhiihuberniuk/blog-api/view/models"
+)
+
+func (h *Handlers) CreatePost(w http.ResponseWriter, r *http.Request) {
+	var in viewmodels.CreatePostRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&in); err != nil {
+		http.Error(w, "cannot decode data from JSON", http.StatusBadRequest)
+
+		return
+	}
+
+	postID, err := h.service.CreatePost(r.Context(), models.CreatePostPayload{
+		Title:       in.Title,
+		Description: in.Description,
+		Tags:        in.Tags,
+		AuthorID:    in.AuthorID,
+	})
+	if err != nil {
+		http.Error(w, "cannot create post", http.StatusInternalServerError)
+
+		return
+	}
+
+	post, err := h.service.GetPost(r.Context(), postID)
+	if err != nil {
+		http.Error(w, "cannot get created post", http.StatusNotFound)
+
+		return
+	}
+
+	out := viewmodels.GetPostResponse{
+		ID:          post.ID,
+		Title:       post.Title,
+		Description: post.Description,
+		CreatedAt:   post.CreatedAt,
+		CreatedBy:   post.CreatedBy,
+		Tags:        post.Tags,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err = json.NewEncoder(w).Encode(out); err != nil {
+		http.Error(w, "cannot encode data into JSON", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func (h *Handlers) GetPost(w http.ResponseWriter, r *http.Request) {
+	postID := mux.Vars(r)["id"]
+
+	post, err := h.service.GetPost(r.Context(), postID)
+	if err != nil {
+		http.Error(w, "cannot find post with such ID", http.StatusNotFound)
+
+		return
+	}
+
+	out := viewmodels.GetPostResponse{
+		ID:          post.ID,
+		Title:       post.Title,
+		Description: post.Description,
+		CreatedAt:   post.CreatedAt,
+		CreatedBy:   post.CreatedBy,
+		Tags:        post.Tags,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err = json.NewEncoder(w).Encode(out); err != nil {
+		http.Error(w, "cannot encode data into JSON", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func (h *Handlers) UpdatePost(w http.ResponseWriter, r *http.Request) {
+	postID := mux.Vars(r)["id"]
+
+	post, err := h.service.GetPost(r.Context(), postID)
```

Comment:
in all update operation you don't need to get existing model and initialize "in" model with it